cobra/core/Model.py

```diff
@@ -230,9 +308,85 @@ def add_reactions(self, reaction_list):
 
         self.reactions += reaction_list
 
+        # from cameo ...
+        self._populate_solver(reaction_list)
+
+    def _populate_solver(self, reaction_list, metabolite_list=None):
+        """Populate attached solver with constraints and variables that
+        model the provided reactions.
+        """
+        constraint_terms = AutoVivification()
+        if metabolite_list is not None:
```

Comment:
But why. The solver will ignore all zero rows anyways. It just gives the presolver a little more work. If it eventually does participate in a reaction this will trigger _populate_solver again which will add the constraint in the first reaction where it is used. Metabolites that do not participate in any reaction should not affect the solver at all...